central/externalbackups/plugins/s3/s3.go

```diff
@@ -120,24 +164,25 @@ func sortS3Objects(objects []*awsS3.Object) {
 	})
 }
 
-func (s *s3) pruneBackupsIfNecessary() error {
-	objects, err := s.svc.ListObjects(&awsS3.ListObjectsInput{
+func (s *s3) pruneBackupsIfNecessary(ctx context.Context) error {
+	listedBackups, err := s.awsClient.ListObjectsV2(ctx, &sdkS3.ListObjectsV2Input{
 		Bucket: aws.String(s.integration.GetS3().GetBucket()),
 		Prefix: aws.String(s.prefixKey("backup")),
 	})
 	if err != nil {
 		return s.createError(fmt.Sprintf("listing objects in s3 bucket %q", s.bucket), err)
 	}
-	sortS3Objects(objects.Contents)
 
-	var objectsToRemove []*awsS3.Object
-	if len(objects.Contents) > int(s.integration.GetBackupsToKeep()) {
-		objectsToRemove = objects.Contents[s.integration.GetBackupsToKeep():]
+	sortS3Objects(listedBackups.Contents)
+
+	var objectsToRemove []s3Types.Object
+	if len(listedBackups.Contents) > int(s.integration.GetBackupsToKeep()) {
+		objectsToRemove = listedBackups.Contents[s.integration.GetBackupsToKeep():]
 	}
 
-	errorList := errorhelpers.NewErrorList("remove objects in S3 store")
+	errorList := errorhelpers.NewErrorList("remove objects in s3 store")
```

Comment:
In other places its capital
```suggestion
	errorList := errorhelpers.NewErrorList("remove objects in S3 store")
```